action_plugins/cli.py

```diff
@@ -105,6 +105,14 @@ def run(self, tmp=None, task_vars=None):
         socket_path = getattr(self._connection, 'socket_path') or task_vars.get('ansible_socket')
         connection = Connection(socket_path)
 
+        # make command a required argument
+        if not command:
+            raise AnsibleError('missing required argument `command`')
+
+        # when default value is set to None Ansible sets it to ''
+        if not engine:
```

Comment:
@privateip when `engine` is not set in playbook, it is set to `None` because of `network_engine_engine: "{{ engine | default(None) }}"` in defaults/main.yaml when `cli` task is used.
what `engine = self._task.args.get('engine', 'command_parser')` does is, it checks if the key `engine` exists, it not exists it set it to `command_parser`.

since defaults/main.yaml sets None to engine as default, task gets a key `engine` with value `''`.
So either we need to update the defaults/main.yaml to `network_engine_engine: "{{ engine | default(command_parser) }}"` or have this check.

Reply:
The change should be made to remove setting the a value in `defaults/` ... My bad for not catching that sooner.